indico/modules/events/abstracts/util.py

```diff
@@ -63,42 +79,82 @@ def _get_static_columns(self, ids):
 
         :return: a list of {'id': ..., 'caption': ...} dicts
         """
-        return [{'id': id_, 'caption': self.static_items[id_]['title']} for id_ in ids if id_ in self.static_items]
+        return [{'id': id_, 'caption': self.static_items[id_]['title']} for id_ in self.static_items if id_ in ids]
+
+    def _get_sorted_contribution_fields(self, item_ids):
+        """Return the contribution fields ordered by their position in the abstract form."""
+
+        return (ContributionField
+                .find(ContributionField.id.in_(item_ids))
+                .with_parent(self.list_event)
+                .order_by(ContributionField.position)
+                .all())
+
+    def _get_filters_from_request(self):
+        filters = super(AbstractListGenerator, self)._get_filters_from_request()
+        for field in self.list_event.contribution_fields:
+            if field.field_type == 'single_choice':
+                options = request.form.getlist('field_{}'.format(field.id))
+                if options:
+                    filters['fields'][str(field.id)] = options
```

Comment:
most likely stored as json where all dict keys become strings. unicode instead of str is just nicer since we use unicode everywhere anyway